src/systems/collada_world_exporter/ColladaWorldExporter.cc

```diff
@@ -0,0 +1,261 @@
+
+          addSubmeshFunc(i);
+        }
+      }
+      else if (_geom->Data().Type() == sdf::GeometryType::CYLINDER)
+      {
+        if (meshManager->HasMesh("unit_cylinder"))
+        {
+          mesh = meshManager->MeshByName("unit_cylinder");
+          scale.X() = _geom->Data().CylinderShape()->Radius() * 2;
+          scale.Y() = scale.X();
+          scale.Z() = _geom->Data().CylinderShape()->Length();
+
+          int i = worldMesh.AddMaterial(mat);
+
+          addSubmeshFunc(i);
+        }
+      }
+      else if (_geom->Data().Type() == sdf::GeometryType::PLANE)
+      {
+        if (meshManager->HasMesh("unit_plane"))
+        {
+          // Create a rotation for the plane mesh to account
+          // for the normal vector.
+          mesh = meshManager->MeshByName("unit_plane");
+
+          scale.X() = _geom->Data().PlaneShape()->Size().X();
+          scale.Y() = _geom->Data().PlaneShape()->Size().Y();
+
+          // // The rotation is the angle between the +z(0,0,1) vector and the
+          // // normal, which are both expressed in the local (Visual) frame.
+          math::Vector3d normal = _geom->Data().PlaneShape()->Normal();
+          math::Quaterniond normalRot;
+          normalRot.From2Axes(math::Vector3d::UnitZ, normal.Normalized());
+          worldPose.Rot() = worldPose.Rot() * normalRot;
+
+          matrix = math::Matrix4d(worldPose);
+
+          int i = worldMesh.AddMaterial(mat);
+          addSubmeshFunc(i);
+        }
+      }
+      else if (_geom->Data().Type() == sdf::GeometryType::SPHERE)
+      {
+        if (meshManager->HasMesh("unit_sphere"))
+        {
+          mesh = meshManager->MeshByName("unit_sphere");
+
+          scale.X() = _geom->Data().SphereShape()->Radius() * 2;
+          scale.Y() = scale.X();
+          scale.Z() = scale.X();
+
+          int i = worldMesh.AddMaterial(mat);
+
+          addSubmeshFunc(i);
+        }
+      }
+      else if (_geom->Data().Type() == sdf::GeometryType::MESH)
+      {
+        auto fullPath = asFullPath(_geom->Data().MeshShape()->Uri(),
+            _geom->Data().MeshShape()->FilePath());
+
+        if (fullPath.empty())
+        {
+          ignerr << "Mesh geometry missing uri" << std::endl;
+          return true;
+        }
+        mesh = meshManager->Load(fullPath);
+
+        if (!mesh) {
+          ignerr << "mesh not found!" << std::endl;
+          return true;
+        }
+
+        for (unsigned int k = 0; k < mesh->SubMeshCount(); k++)
+        {
+          auto subMeshLock = mesh->SubMeshByIndex(k).lock();
+          int j = subMeshLock->MaterialIndex();
+
+          int i = 0;
+          if (j != -1)
+          {
+            i = worldMesh.IndexOfMaterial(mesh->MaterialByIndex(j).get());
+            if (i < 0)
+            {
+              i = worldMesh.AddMaterial(mesh->MaterialByIndex(j));
+            }
+          }
+          else
+          {
+            i = worldMesh.AddMaterial(mat);
+          }
+
```

Comment:
```suggestion
scale =  _geom->Data().MeshShape()->Scale();
```

Reply:
Added in e44a66efca495ce50beba9ad1b4d173d690ba9be